src/components/button/button.spec.tsx

```diff
@@ -36,12 +38,41 @@ const sizesHeights: [SizeOptions, string][] = [
   ["medium", "40px"],
   ["large", "48px"],
 ];
+
+describe("ButtonWithForwardRef", () => {
+  it("should display deprecation warning when the component is used once", () => {
+    const loggerSpy = jest.spyOn(Logger, "deprecate");
+
+    const wrapper = mount(<ButtonWithForwardRef>Button</ButtonWithForwardRef>);
+
+    expect(loggerSpy).toHaveBeenCalledWith(
+      "The `ButtonWithForwardRef` component is deprecated and will soon be removed. Please use a `Button` component with the `ref` prop instead."
+    );
+
+    wrapper.setProps({ prop1: true });
+    expect(loggerSpy).toHaveBeenCalledTimes(1);
+    loggerSpy.mockRestore();
+  });
+});
+
 describe("Button", () => {
-  describe("refs", () => {
+  it("should display deprecation warning when the forwardRef prop is used", () => {
```

Comment:
it's not a big deal but I'm surprised you're not testing that the deprecation message is only shown on first render, as you did with the one above